src/main/java/response/HttpResponseWriter.java

```diff
@@ -1,18 +1,28 @@
 package com.td.HttpServer;
-
 import java.util.*;
 import java.io.*;
 
-public class HttpResponseFormatter {
+public class HttpResponseWriter {
 
   private final byte[] NEWLINE = "\r\n".getBytes();
   private HashMap<Integer, String> responseLines;
 
-  public HttpResponseFormatter() {
+  public HttpResponseWriter() {
     buildResponseLineMap();
   }
 
-  public byte[] responseAsBytes(HttpResponse response) {
+  public void sendHttpResponse(IClientSocketIO client, HttpResponse response)throws BadConnectionException {
+    byte[] responseBytes = responseAsBytes(response);
+    try {
+      client.sendBytes(responseBytes);
+    }
+    catch (IOException e) {
+      e.printStackTrace();
+      throw new BadConnectionException();
```

Comment:
One last thing about exceptions - it's a good idea to wrap exceptions so that, in case you need do need it (because you're debugging or reading error logs, _not_ because the layer above you wants to get it's hands on it programmatically).

`new BadConnectionException(e)`

That _should_ work out of the box because deriving from `Exception` will get you those constructors free of charge.